### PR TITLE
Tweaked Hydra flow triggers for onprem

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
If minimum tests executed or if override results directory feature is used the task will redirected to the fully managed code flow.

### CODE REVIEW

File: Tasks/VsTestV2/runvstest.ts [hunk 1]
```diff
@@ -78,6 +79,39 @@ async function execute() {
     }
 }
 
+function isHydraFlowToBeEnabled(inputDataContract: InputDataContract) {
+    if ((inputDataContract.ServerType && inputDataContract.ServerType.toLowerCase() === 'hosted')) {
```
Comment (phanikmmsft) [inline Tasks/VsTestV2/runvstest.ts:83]: suggest using constants for these hardcoded strings

File: Tasks/VsTestV2/runvstest.ts [hunk 2]
```diff
@@ -78,6 +79,39 @@ async function execute() {
...
+
+        tl.debug('Enabling Hydra flow since serverType is hosted.');
+        return true;
+    }
+
+    if (tl.getVariable('Force_Hydra') && tl.getVariable('Force_Hydra').toLowerCase() === 'true') {
```
Comment (phanikmmsft) [inline Tasks/VsTestV2/runvstest.ts:89]: Is this a hidden /backdoor variable you are proposing?

File: Tasks/VsTestV2/runvstest.ts [hunk 3]
```diff
@@ -78,6 +79,39 @@ async function execute() {
...
+    if (inputDataContract.TestReportingSettings
+        && !utils.Helper.isNullEmptyOrUndefined(inputDataContract.TestReportingSettings.TestResultsDirectory)
+        && inputDataContract.TestReportingSettings.TestResultsDirectory.toLowerCase()
+            !== path.join(tl.getVariable('Agent.TempDirectory'), 'TestResults').toLowerCase()) {
+
+        tl.debug('Enabling Hydra flow since the override results directory feature is being used.');
+        return true;
+    }
+
+    return false;
```
Comment (phanikmmsft) [inline Tasks/VsTestV2/runvstest.ts:112]: consider wrapping this in try / catch and return false if there is any errors with any of these condition checks

File: Tasks/VsTestV2/runvstest.ts (hunk 2 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/runvstest.ts:89]: Yes

File: Tasks/VsTestV2/runvstest.ts (hunk 3 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/runvstest.ts:112]: fair point

**Pull request merged**